### PR TITLE
Margin is valid with no base

### PR DESCRIPTION
**Pull request by raymogg**
# Motivation
There was a bug with how withdraw was computing `marginIsValid`. This allowed a user to withdraw more than deposited when they had no base (positions) outstanding.

# Changes
- ensure quote > 0 if base = 0 in `marginIsValid`
- removes redundant `margin < 0` check

### CODE REVIEW

File: contracts/TracerPerpetualSwaps.sol [hunk 1]
```diff
@@ -453,13 +453,10 @@ contract TracerPerpetualSwaps is
             Balances.minimumMargin(pos, price, gasCost, maxLeverage);
         int256 margin = Balances.margin(pos, price);
 
-        if (margin < 0) {
-            /* Margin being less than 0 is always invalid, even if position is 0.
-               This could happen if user attempts to over-withdraw */
-            return false;
-        }
```
Comment (CalabashSquash) [inline contracts/TracerPerpetualSwaps.sol]: Shouldn't this still be in anyway?

File: contracts/TracerPerpetualSwaps.sol [hunk 2]
```diff
@@ -453,13 +453,10 @@ contract TracerPerpetualSwaps is
...
         if (minMargin == 0) {
-            return true;
+            // minMargin = 0 only occurs when user has no base (positions)
+            // if they have no base, their quote must be > 0.
```
Comment (sporejack) [inline contracts/TracerPerpetualSwaps.sol:494]: Mismatch between comment and actual statement below. I'm assuming the comment is wrong and should be inclusive?

Comment (sporejack) [reply contracts/TracerPerpetualSwaps.sol:494]: I.e.,

```suggestion
            // if they have no base, their quote must be >= 0.
```

File: contracts/TracerPerpetualSwaps.sol (hunk 1 above)
Comment (raymogg) [reply contracts/TracerPerpetualSwaps.sol]: @BenjaminPatch `Balances.minimumMargin` now returns a uint so it cant be less than 0. I think this existed as previously we returned a int and then casted so we needed to make this safety check.

Comment (CalabashSquash) [reply contracts/TracerPerpetualSwaps.sol]: This is margin though, not minimum margin

**Review by sporejack — CHANGES_REQUESTED** [review]
Might be a controversial opinion, but should we just convert the body of `marginIsValid` into a ternary?

File: contracts/TracerPerpetualSwaps.sol (hunk 2 above)
Comment (raymogg) [reply contracts/TracerPerpetualSwaps.sol:494]: @sporejack are you giving me permission to make a triple chained ternary? 👀

**Review by CalabashSquash — APPROVED** [review]
:sri_lanka:

File: contracts/TracerPerpetualSwaps.sol (hunk 2 above)
Comment (sporejack) [reply contracts/TracerPerpetualSwaps.sol:494]: @raymogg Not anymore!

**Pull request merged**